transforms/unquote-properties.js

```diff
@@ -0,0 +1,33 @@
+/**
+ * Removes quotes from object properties whose keys are strings which are valid
+ * identifiers. Does not handle quoted methods.
+ */
+module.exports = (file, api, options) => {
+  const j = api.jscodeshift;
+  const printOptions = options.printOptions || {quote: 'single'};
+  const root = j(file.source);
+
+  const isValidIdentifierNameForProperty = (name) => {
+    try {
+      new Function(`({${name}: 1})`)(); //eslint-disable-line no-new-func
```

Comment:
for the record: this is a smart meta-evaluation way that doesn't seem harmful. I'm going to merge this because it is a good example of how meta-evaluation can potentially be useful in codemods. In this particular case, a regex is probably smarter.